ambition_utils/rrule/models.py

```diff
@@ -78,18 +100,29 @@ def process_related_model_handlers(self):
 
         return rrules_to_advance
 
-    def overdue_handler_class_instances(self, **kwargs):
+    def overdue_handler_class_instances(self, limit=None, **kwargs):
         """
         Returns a set of instances for any handler with an old next_occurrence
         """
 
-        # Get the rrule objects that are overdue and need to be handled
+        # Get overdue rrules where the next_occurrence is the earliest for each occurrence_handler_path
         rrule_objects = self.get_queryset().filter(
             next_occurrence__lte=datetime.utcnow(),
             **kwargs
-        ).distinct(
+        ).annotate(
+            row_num=Window(
+                expression=RowNumber(),
+                partition_by=F('occurrence_handler_path'),
+                order_by=F('next_occurrence').asc()
+            )
+        ).filter(
+            row_num=1
```

Comment:
so this is the part that restricts the rrule handler from processing more than one recurrence type in a single tick?

Reply:
this just ensures proper ordering in the event we are running with a limit, we want to prioritize a handler that has the most out of date occurrence